Add document-link elements within iati-activity/result

dalepotter opened the pull request:
Fixes #359

dalepotter committed:
Add iati-activity/result/document-link and test cases

dalepotter committed:
Modify definition of iati-activity/result/document-link in line with proposal

allthatilk commented on tests/should-fail/iati-activities/iati-activity/result/document-link/description/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml:
```diff
@@ -0,0 +1,41 @@
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <description additional-attribute="xx"><!-- FAIL: An addiitonal attribute added. -->
```
typo: addiitonal -> `additional`

allthatilk commented on tests/should-fail/iati-activities/iati-activity/result/document-link/category/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml:
```diff
@@ -0,0 +1,38 @@
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <category code="xx" additional-attribute="xx" /><!-- FAIL: An addiitonal attribute added. -->
```
typo on additional here also

allthatilk commented on tests/should-fail/iati-activities/iati-activity/result/document-link/description/err-not-iati-xml-missing-required-element_document-link-with-multiple-description-elements.xml:
```diff
@@ -0,0 +1,44 @@
+        <description>
+          <narrative>Sample text</narrative>
+        </description>
+        <description><!-- FAIL: Optional element more than the allowed number of times. -->
```
rephrase: `Optional element present more than the permitted number of times.`

allthatilk commented on tests/should-fail/iati-activities/iati-activity/result/document-link/document-date/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml:
```diff
@@ -0,0 +1,39 @@
+          <narrative>Sample text</narrative>
+        </title>
+        <category code="xx"/>
+        <document-date iso-date="xx" additional-attribute="xx"/><!-- FAIL: An additional attribute added. -->
```
Rephrase: `Non-permitted additional attribute present.`

allthatilk commented on tests/should-fail/iati-activities/iati-activity/result/document-link/document-date/err-not-iati-xml-incorrect-datatype.xml:
```diff
@@ -0,0 +1,39 @@
+          <narrative>Sample text</narrative>
+        </title>
+        <category code="xx"/>
+        <document-date iso-date="xx"/><!-- FAIL: Not an XML date type. -->
```
rephrase: `Not a valid xsd:date.`

allthatilk commented on tests/should-fail/iati-activities/iati-activity/result/document-link/document-date/err-not-iati-xml-missing-required-element_multiple-elements-present.xml:
```diff
@@ -0,0 +1,40 @@
+        </title>
+        <category code="xx"/>
+        <document-date iso-date="2000-01-01"/>
+        <document-date iso-date="2000-01-02"/><!-- FAIL: An addiitonal element is added. -->
```
The typo returns! addiitonal -> `additional`

allthatilk commented on tests/should-fail/iati-activities/iati-activity/result/document-link/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml:
```diff
@@ -0,0 +1,38 @@
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx" additional-attribute="xx"><!-- FAIL: Optional document-link element added with required child attributes/elements but an addiitonal attribute added. -->
```
and here too `additional`

allthatilk commented on tests/should-fail/iati-activities/iati-activity/result/document-link/language/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml:
```diff
@@ -0,0 +1,39 @@
+          <narrative>Sample text</narrative>
+        </title>
+        <category code="xx"/>
+        <language additional-attribute="xx"/><!-- FAIL: An addiitonal attribute added. -->
```
And here

allthatilk commented on tests/should-fail/iati-activities/iati-activity/result/document-link/title/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml:
```diff
@@ -0,0 +1,38 @@
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx">
+        <title additional-attribute="xx"><!-- FAIL: An addiitonal attribute added. -->
```
and here

allthatilk commented on tests/should-fail/iati-activities/iati-activity/result/document-link/title/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml:
```diff
@@ -0,0 +1,38 @@
+        <title additional-attribute="xx"><!-- FAIL: An addiitonal attribute added. -->
+          <narrative>Sample text</narrative>
+        </title>
+        <category code="xx" />
```
remove unnecessary whitespace

dalepotter replied:
Is there unnecessary whitespace - I can't see any...

![image](https://user-images.githubusercontent.com/8247168/33430945-42ead668-d5ca-11e7-921a-1983f200f4b7.png)

allthatilk commented on tests/should-fail/iati-activities/iati-activity/result/document-link/title/narrative/err-not-iati-xml-missing-required-element_no-narrative-element.xml:
```diff
@@ -0,0 +1,38 @@
+      </title>
+      <document-link url="xx" format="xx">
+        <title>
+          <!-- FAIL: Required `title` element added, but without the required 'narrative' child element. -->
```
Remove whitespace

dalepotter replied:
Is the suggestion that this should be:

```
<title/><!-- FAIL: Required `title` element added, but without the required 'narrative' child element. -->
```

??

allthatilk commented on tests/should-pass/iati-activities/iati-activity/result/document-link/document-date/element-present-with-namespace-attribute.xml:
```diff
@@ -0,0 +1,39 @@
+          <narrative>Sample text</narrative>
+        </title>
+        <category code="xx"/>
+        <document-date iso-date="2000-01-01" example-namespace:namespace-attribute="xx"/><!-- PASS: Optional element added with required child element, and namespace attribute. -->
```
remove comma before `and`

allthatilk commented on tests/should-pass/iati-activities/iati-activity/result/document-link/element-present-with-namespace-attribute.xml:
```diff
@@ -0,0 +1,38 @@
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx" example-namespace:namespace-attribute="xx"><!-- PASS: Optional element added with required child elements, and namespace attribute. -->
```
remove comma before `and` here and elsewhere

allthatilk commented on tests/should-pass/iati-activities/iati-activity/result/document-link/language/element-present-with-namespace-attribute.xml:
```diff
@@ -0,0 +1,39 @@
+          <narrative>Sample text</narrative>
+        </title>
+        <category code="xx"/>
+        <language code="xx" example-namespace:namespace-attribute="xx"/><!-- PASS: Optional element added with required child element, and namespace attribute. --> 
```
here

allthatilk commented on tests/should-pass/iati-activities/iati-activity/result/document-link/category/element-present-with-namespace-attribute.xml:
```diff
@@ -0,0 +1,38 @@
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <category code="xx" example-namespace:namespace-attribute="xx"/><!-- PASS: Element added with required child element, and namespace attribute. -->
```
here

dalepotter replied:
what's here? Guessing the unnecessary comma?

allthatilk commented on tests/should-pass/iati-activities/iati-activity/result/document-link/description/element-present-with-namespace-attribute.xml:
```diff
@@ -0,0 +1,41 @@
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <description example-namespace:namespace-attribute="xx"><!-- PASS: Optional element added with required child element, and namespace attribute. -->
```
here

allthatilk commented on tests/should-pass/iati-activities/iati-activity/result/document-link/description/narrative/element-present-with-namespace-attribute.xml:
```diff
@@ -0,0 +1,41 @@
+          <narrative>Sample text</narrative>
+        </title>
+        <description>
+          <narrative example-namespace:namespace-attribute="xx">Sample text</narrative><!-- PASS: Required element added, and namespace attribute. -->
```
here

allthatilk commented on tests/should-pass/iati-activities/iati-activity/result/document-link/title/element-present-with-namespace-attribute.xml:
```diff
@@ -0,0 +1,38 @@
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx">
+        <title example-namespace:namespace-attribute="xx"><!-- PASS: Optional element added with required child element, and namespace attribute. -->
```
here

allthatilk commented on tests/should-pass/iati-activities/iati-activity/result/document-link/title/narrative/element-present-with-namespace-attribute.xml:
```diff
@@ -0,0 +1,41 @@
+      </title>
+      <document-link url="xx" format="xx">
+        <title>
+          <narrative example-namespace:namespace-attribute="xx">Sample text</narrative><!-- PASS: Required element added, and namespace attribute. -->
```
here

allthatilk commented on tests/should-pass/iati-activities/iati-activity/result/document-link/title/narrative/multiple-elements-present.xml:
```diff
@@ -0,0 +1,42 @@
+      <document-link url="xx" format="xx">
+        <title>
+          <narrative>Sample text</narrative>
+          <narrative>Sample text</narrative><!-- PASS: Multiple elements present. -->
```
rephrase: `element present multiple times in context`

allthatilk reviewed (changes requested):
All just small typo things

dalepotter committed:
Fix typos and unnecessary commas in test case files.

dalepotter committed:
Remove additional unnecessary commas in test case files.

dalepotter committed:
Remove unneccessary whitespace in test case files.

dalepotter committed:
Add iati-activity/result/indicator/document-link and test cases

dalepotter committed:
Remove unnecessary whitespace character in test case files.

dalepotter committed:
Add iati-activity/result/indicator/baseline/document-link and test cases

dalepotter committed:
Add iati-activity/result/indicator/period/target/document-link and test cases

dalepotter committed:
Add iati-activity/result/indicator/period/actual/document-link and test cases

allthatilk commented on tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/language/element-present.xml:
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0"?>
+
+<iati-activities version="xx">
+  <iati-activity xmlns:example-namespace="http://example.org/example-namespace/ns#"><!-- Include namespace declaration. -->
```
should this be here?

dalepotter replied:
Fixed in 4a91fd0

allthatilk commented on tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/document-date/element-present.xml:
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0"?>
+
+<iati-activities version="xx">
+  <iati-activity xmlns:example-namespace="http://example.org/example-namespace/ns#"><!-- Include namespace declaration. -->
```
here?

dalepotter replied:
Fixed in 4a91fd0

allthatilk commented on tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/title/element-present-with-namespace-attribute.xml:
```diff
@@ -0,0 +1,38 @@
+          <title example-namespace:namespace-attribute="xx"><!-- PASS: Optional element added with required child element and namespace attribute. -->
+            <narrative>Sample text</narrative>
+          </title>
+          <category code="xx" />
```
whitespace, might want to do a check throughout as it always seems to be with the code attribute.

dalepotter replied:
This was resolved in caf1ce7.

allthatilk commented on tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/title/element-present-with-namespace-subelement.xml:
```diff
@@ -0,0 +1,39 @@
+            <narrative>Sample text</narrative>
+            <example-namespace:namespace-element></example-namespace:namespace-element> <!-- PASS: Optional namespace element included. -->
+          </title>
+          <category code="xx" />
```
whitespace

dalepotter replied:
This was resolved in caf1ce7.

dalepotter committed:
Remove unnecessary namespace declaration from test case files

dalepotter committed:
Fix schema ordering of new document-link elements

This fix is made in line with the comments at: https://github.com/IATI/IATI-Schemas/issues/359#issuecomment-349931362

dalepotter committed:
Merge branch 'result-element-integration' into 2.03-add-document-link-to-results-elements

dalepotter commented:
@allthatilk this is now ready for re-review (checking the three most recent commits to this branch should suffice).
